src/main/java/com/guciowons/shoppingbasket/Product/Product.java

```diff
@@ -1,18 +1,38 @@
 package com.guciowons.shoppingbasket.Product;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
 import java.math.BigDecimal;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 
+@Document
 public class Product {
 
+    @Id
+    private String appId;
     private Integer id;
     private String title;
     private BigDecimal price;
     private String description;
     private String category;
     private String image;
     private Rating rating;
-    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
+    private Map<String, Object> additionalProperties = new HashMap<>();
```

Comment:
To jest na pewno potrzebne aktualnie, żeby zapisywać w twojej bazie? Jeśli to jest dodane tylko dlatego, żeby być zgodnym z zewnętrzym API to tak nie może być. Stwórz obiekt, który będzie odzwierciedlał Produkt z tamtego API, przechwytuj go z API i tłumacz go na twój obiekt Product, który będzie zawierał tylko wymagane i potrzebne wartości.